src/Ocelot/DependencyInjection/ServiceCollectionExtensions.cs

```diff
@@ -17,15 +20,15 @@ public static class ServiceCollectionExtensions
     /// <returns>An <see cref="IOcelotBuilder"/> object.</returns>
     public static IOcelotBuilder AddOcelot(this IServiceCollection services)
     {
-        var configuration = services.BuildServiceProvider().GetRequiredService<IConfiguration>();
+        var configuration = services.FindConfiguration(null);
         return new OcelotBuilder(services, configuration);
     }
```

Comment:
`[Obsolete]` ?

Reply:
@ArwynFr @raman-m Mhh, well, the point of @ArwynFr was, using IConfiguration should be regarded as obsolete, not the opposite. We would then inject the settings using the options pattern, eg. 
```c#
builder.Services.Configure<PositionOptions>(
    builder.Configuration.GetSection(PositionOptions.Position));
```
... And then DI
```c#
public Test2Model(IOptions<PositionOptions> options)
```
https://learn.microsoft.com/en-us/aspnet/core/fundamentals/configuration/options?view=aspnetcore-8.0

Reply:
Now, the problem is, the Ocelot Pipeline isn't supporting the options pattern yet. So, what we are doing here is a bit artificial, but it works.

Reply:
🆗 Guys, if you don't want to put `[Obsolete]` attribute, it is OK.  

> If we deprecate then undeprecate it, users will be confused.

OK! Got it! Not an issue!